sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -134,9 +134,42 @@ object ResolveHints {
    * This must be executed after all the other hint rules are executed.
    */
   object RemoveAllHints extends Rule[LogicalPlan] {
-    def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperatorsUp {
-      case h: UnresolvedHint => h.child
+
+    def apply(plan: LogicalPlan): LogicalPlan = {
+      val ignoredHintLogger = new IgnoredHintLogger()
+      plan resolveOperatorsUp {
+        case h: UnresolvedHint =>
+          // Logs the unused plan hint before we remove it
+          ignoredHintLogger.log(h)
+          h.child
+      }
     }
-  }
 
+    private class IgnoredHintLogger {
```

Comment:
Why do we need a new class here? Is there any reason why we can't use `HintErrorLogger`?

Reply:
Yea, HintErrorLogger can covert all the case in this pr, so I'll close this. Thanks.